src/main/java/uk/gov/ida/matchingserviceadapter/saml/UserIdHashFactory.java

```diff
@@ -4,47 +4,51 @@
 import org.apache.commons.codec.binary.Hex;
 import org.opensaml.security.crypto.JCAConstants;
 import uk.gov.ida.saml.core.domain.AuthnContext;
-import uk.gov.ida.saml.core.domain.IdentityProviderAuthnStatement;
 
 import java.io.UnsupportedEncodingException;
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 import java.text.MessageFormat;
 
-import static com.google.common.base.Throwables.propagate;
-
 public class UserIdHashFactory {
-    public String createHashedId(String partnerEntityId, String entityId, String persistentId, Optional<IdentityProviderAuthnStatement> authnStatement) {
+
+    private final String msaEntityId;
+
+    public UserIdHashFactory(String msaEntityId) {
+        this.msaEntityId = msaEntityId;
+    }
+
+    public String hashId(String issuerEntityId, String persistentId, Optional<AuthnContext> authnContext) {
```

Comment:
```AuthnContext ``` has been deprecated. We should duplicate the class in the MSA instead.

Reply:
Will probably try to make the other changes and get this merged and then come back and look at all the deprecated classes.

Reply:
👍